### DISCUSSION TITLE
Export the factory.factory struct

### DISCUSSION THREAD

**Discussion by jacobmolby** [stephenafamo/bob#57] [Ideas]
Hi, is there a specific reason for why the `factory` struct isn't exported by the package?

I think there is a lot of valid use cases for having it exported. 

For instance if you have a lot of mods that you need to apply to the factory in your tests, you could have a helper function return a modified factory instance with the mods applied. 

This however isn't possible at the moment. Since the type is private, it leads to a lot duplicated code.

If this were to be changed it should probably considered as a breaking change. If you are already importing another package that has a Factory struct inside it, it would conflict.

**Comment by stephenafamo** [comment]
There is no specific reason, I just tend to be quite conservative with exporting things to reduce the need to introduce any breaking changes.

Making the `factory.Factory` exported is a breaking change, but it is unlikely to conflict with anything else.

**Reply by stephenafamo** [reply]
Try d3082ec

**Reply by jacobmolby** [reply]
Hi sorry for not replying sooner, but it looks good. 

Thank you for being so active and fast,
